high_card.rb

```diff
@@ -2,4 +2,67 @@
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
+cards = []
+
+ranks.each do |rank|
+  suits.each do |suit|
+    cards.push([rank,suit])
+  end
+end
+
+cards = cards.shuffle
+
 players = []
+players_cards = []
+
+puts 'How many players are playing today?'
+num_players = gets.chomp.to_i
+
+# question - is it best practice to always pass arugments to fuctions/methods
+# rather than using glabal variables (both JS and ruby)?
+# didnt get it to work without passing in arguments
+def play_game players,cards
+  players.each_with_index do |player, index|
+    player[:card] = cards[index]
+    case player[:card][0]
+    when "A"
+      player[:card][0] = 13
+    when "J"
+      player[:card][0] = 11
+    when "Q"
+      player[:card][0] = 12
+    when "K"
+      player[:card][0] = 13
+    else
+    end
+  end
+  puts players
+
+  max = players.max { |a, b| a[:card][0] <=> b[:card][0]}
+  winners_array = players.find_all do |player|
+    player[:card][0] == max[:card][0]
+  end
+  return winners_array
+end
+
+counter = 1
+while(counter <= num_players) do
+  puts "#{players.length} players so far. Please enter a player name, or type 'play' to start playing"
+  name = gets.chomp
+  if name == 'play'
```

Comment:
I'd say pick either using `if name == 'play' then break` or `while(counter <= num_players)`... If you want the game to start whenever the player types "play", then you don't really need to ask/measure the number of players.